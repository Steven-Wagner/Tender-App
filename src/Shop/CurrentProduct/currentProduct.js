import React from 'react';
import Stars from '../../Components/Stars/stars';

function CurrentProduct(props) {
    return(
        <section className='current-product'>
            <div className='choose-buttons'>
                <button onClick={() => props.handleSkip()}>Skip</button>
                <button onClick={() => props.handleBuy()}>Buy</button>
            </div>
            {/* <button onClick={() => props.handleLike()}>Like</button> */}            <h2>{props.item.title}</h2>
            <img className='product-pic' src={props.item.img} alt={props.item.title}/>
            <Stars rating={props.item.rating}/>
            <p>Price: {props.item.price}</p>
            <p>{props.item.description}</p>
        </section>
    )
}

export default CurrentProduct;