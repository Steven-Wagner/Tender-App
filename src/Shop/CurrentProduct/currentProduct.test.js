import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import CurrentProduct from './currentProduct';
import Enzyme, { shallow, mount, render, setState, update } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import { library } from '@fortawesome/fontawesome-svg-core';
// import { faCoins } from '@fortawesome/free-solid-svg-icons';
import ReactDOM from 'react-dom';

// library.add(faCoins);

Enzyme.configure({ adapter: new Adapter() });

const item = {
    title: 'Current Product', 
    img: 'Current img', 
    desription: 'Current Product description', 
    sold: '2', 
    profit: '4', 
    price: '2',
    index: 0};

function makeCurrentProductWrapper(item) {
    return mount(
        <Router>
            <CurrentProduct
                item={item}/>
        </Router>)
}

describe('Current Product UI renders correctly', () => {
    it('renders content without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <Router>
                <CurrentProduct
                    item={item}/>
            </Router> , div);
    });
      
    it('Shop renders correctly', () => {
        const tree = makeCurrentProductWrapper(item);
        expect(tree).toMatchSnapshot();
        tree.unmount();
    });

    describe('img displays', () => {
        const imgContent = [{display: true, src: 'An img'}, {display: false, src: ''}]

        imgContent.forEach(img => {
            it(`Img displays ${img.display} if item.img ${img.display ? 'has content' : 'is undefined'}`, () => {
                const imgItem = Object.assign({}, item);
                imgItem.img = img.src;
                const wrapper = makeCurrentProductWrapper(imgItem);
                expect(wrapper.exists('.product-pic')).toBe(img.display)
                wrapper.unmount();
            })
        })
    })

    describe('Renders no more products to view', () => {
        it('When no products to show', () => {
            const emptyItem = {index: -1};

            const wrapper = makeCurrentProductWrapper(emptyItem);
            
            expect(wrapper.exists('.no-products')).toBe(true);
            wrapper.unmount();
        })
        it('Renders ad product even if no products in shoppingItems', () => {
            const adItem = Object.assign({}, item);
            delete adItem.index
            adItem.ad = true

            const wrapper = makeCurrentProductWrapper(adItem);
            expect(wrapper.exists('.no-products')).toBe(false);
            wrapper.unmount();
        })
    })
})