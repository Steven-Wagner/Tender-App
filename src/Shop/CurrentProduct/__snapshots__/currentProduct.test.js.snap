// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current Product UI renders correctly Shop renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CurrentProduct
      item={
        Object {
          "desription": "Current Product description",
          "img": "Current img",
          "index": 0,
          "price": "2",
          "profit": "4",
          "sold": "2",
          "title": "Current Product",
        }
      }
    >
      <section
        className="current-product"
      >
        <div
          className="choose-buttons"
        >
          <button
            onClick={[Function]}
          >
            Skip
          </button>
          <button
            className="buy-button"
            onClick={[Function]}
          >
            Buy
          </button>
        </div>
        <h1
          className="shopping-title"
        >
          Current Product
        </h1>
        <img
          alt="Current Product"
          className="product-pic"
          src="Current img"
        />
        <p
          className="price"
        >
          Price: 
          2
        </p>
        <p
          className="description"
        />
      </section>
    </CurrentProduct>
  </Router>
</BrowserRouter>
`;
